### PR TITLE
Tk objects (cleaning up, renaming, and MHT from standalone jets)

### PR DESCRIPTION
**Pull request by emacdonald16**
Hello all,
I modified the names L1TkJets->L1TkMatchedJets and L1TkFastJets->L1TkJets, in order to distinguish more clearly between the track-matched jets and the standalone jets.
I also modified the MHT/HT producer to include an option for creating an MHT/HT collection for standalone jets.

### CODE REVIEW

**Comment by rekovic** [comment]
Thanks @emacdonald16 

I understand the need and support the idea of names distinguishing type of objects, in this case (a) objects matched with tracks and (b) objects made from tracks.  Here, I want to avoid ambiguity as much as possible. _L1TkX_ up to now meant _L1Track_ object matched with _X_ object. This is the case for _L1TkMu_, _L1TkEle_, _L1TkTau_, ...   In light of that, I propose that we stick to that nomenclature and that we make the following changes:

 *  in case of _CaloJets_ which are matched to _L1Tracks_, and which up to now were called _L1TkJet_, we add "Calo" as opposed to "Matched".  So the change would be:
 `L1TkJet`--> `L1TkCaloJet`
 * in the case of Tracker based object, I suggest that we don't use TkX but instead use TrackerX. So for example:
`L1TkFastJet` --> `L1TrackerJet`, etc.

The above clearly describes the object, preserves the meaning of _L1TkX_ nomenclature, and removes the ambiguity with Tracker-standalone objects.

File: L1Trigger/Configuration/python/SimL1Emulator_cff.py [hunk 1]
```diff
@@ -106,26 +106,28 @@
 phase2_SimL1Emulator += L1TkElectrons
 phase2_SimL1Emulator += L1TkIsoElectrons
 phase2_SimL1Emulator += L1TkPhotons
-phase2_SimL1Emulator += L1TkJets
+phase2_SimL1Emulator += L1TkCaloJets
+phase2_SimL1Emulator += L1TrackerJets
 phase2_SimL1Emulator += L1TkPrimaryVertex
```
Comment (mcepeda) [inline L1Trigger/Configuration/python/SimL1Emulator_cff.py]: I think we need to move the primary vertex creation before the trackerJets

File: L1Trigger/L1TTrackMatch/python/L1TkObjectProducers_cff.py [hunk 2]
```diff
@@ -6,20 +6,24 @@
...
+from L1Trigger.L1TTrackMatch.L1TkEmParticleProducer_cfi import L1TkPhotons
 pL1TkPhotons = cms.Path( L1TkPhotons )
 
-from L1Trigger.L1TTrackMatch.L1TkJetProducer_cfi import L1TkJets
-pL1TkJets = cms.Path( L1TkJets)
+from L1Trigger.L1TTrackMatch.L1TrackerJetProducer_cfi import L1TrackerJets
+pL1TrackerJets = cms.Path( L1TrackerJets)
+
+from L1Trigger.L1TTrackMatch.L1TkCaloJetProducer_cfi import L1TkCaloJets
+pL1TkCaloJets = cms.Path( L1TrkCaloJets)
```
Comment (mcepeda) [inline L1Trigger/L1TTrackMatch/python/L1TkObjectProducers_cff.py]: extra r - change to  "pL1TkCaloJets = cms.Path( L1TkCaloJets)"

**Review by mcepeda — COMMENTED** [review]
Hi,
Emily, thanks a lot! I merged in your changes into my area to follow your naming for the menu trees. 
I am however having some problems with the trackerJets: I create a collection but it is empty running on TTJets (other jet collections arent, and TrackerMET isn't either). I'm debugging on my end in case I introduced a problem (starting from a clean area)- but just in case, does the general recipe need to be modified to produce these jets? Can you check all is fine running on one of the new samples?

File: L1Trigger/Configuration/python/SimL1Emulator_cff.py (hunk 1 above)
Comment (emacdonald16) [reply L1Trigger/Configuration/python/SimL1Emulator_cff.py]: Oh that's a good catch! I'll move that.

File: L1Trigger/L1TTrackMatch/python/L1TkObjectProducers_cff.py (hunk 2 above)
Comment (emacdonald16) [reply L1Trigger/L1TTrackMatch/python/L1TkObjectProducers_cff.py]: Fixed!

**Comment by rekovic** [comment]
@emacdonald16 
If you merge the Tracklet code and then commit the removed files, as in your 
ac362e2, this stays in the history, and makes future merging Tracklet branch ineffective.
I cherry-picked your above commits that don't include Tracklet files (tagged as `l1t-phase2-v2.16.9.1`)

@mcepeda 
Tracklet code has been stable for a while now, so I have decided to merge it in our branch
to facilitate future development. (tagged as `l1t-phase2-v2.16.9`)
Latest tag `l1t-phase2-v2.16.10` has the fix VertexProducer configure to use Tracklet L1TTTracks

**Comment by mcepeda** [comment]
@rekovic - ok, so we can go ahead and move to the new tag v2.16.10? any other changes expected in that tag?

**Comment by rekovic** [comment]
@mcepeda Yes, can move to the new tag v2.16.10.  All other commits will be orthogonal to this PR.
Tag is tested with cmsDriver.py command

```
cmsDriver.py step2 --python_filename=rerun_test.py --no_exec -s L1 --datatier GEN-SIM-DIGI-RAW -n 1 --era Phase2_trigger --eventcontent FEVTDEBUGHLT --filein /store/relval/CMSSW_9_3_7/RelValTTbar_14TeV/GEN-SIM-DIGI-RAW/PU25ns_93X_upgrade2023_realistic_v5_2023D17PU200-v1/10000/0A52EE7F-1E2D-E811-86EB-0242AC130002.root --conditions 100X_upgrade2023_realistic_v1 --beamspot HLLHC14TeV --geometry Extended2023D17 --fileout file:step2_2ev_rerun-L1_slim.root --customise=L1Trigger/L1TNtuples/customiseL1Ntuple.L1NtupleRAWEMUGEN_MC --customise=L1Trigger/Configuration/customiseUtils.DropDepricatedProducts --customise=L1Trigger/Configuration/customiseUtils.L1TrackTriggerTracklet --customise=L1Trigger/Configuration/customiseUtils.DropOutputProducts --customise_commands "process.FEVTDEBUGHLToutput.compressionLevel = cms.untracked.int32(2)"
```

You can add ` --no_output ` if you don't need the reprocessed file.

File: L1Trigger/L1TTrackMatch/plugins/L1TrackerJetProducer.cc [hunk 3]
```diff
@@ -144,57 +143,58 @@ void L1TrackerJetProducer::produce(edm::Event& iEvent, const edm::EventSetup& iS
...
+      bool tmp_isPS = false;
+      if (detId.det() == DetId::Detector::Tracker) {
+        if (detId.subdetId() == StripSubdetector::TOB && tTopo->tobLayer(detId) <= 3)     tmp_isPS = true;
+        else if (detId.subdetId() == StripSubdetector::TID && tTopo->tidRing(detId) <= 9) tmp_isPS = true;
+      }
+      if (tmp_isPS) tmp_trk_nstubPS++;
+    }
+    if(tmp_trk_nstubPS<TRK_NSTUBPSMIN)continue;
+    double DeltaZtoVtx=fabs(L1TkPrimaryVertexHandle->begin()->getZvertex()-iterL1Track->getPOCA(L1Tk_nPar).z());
+    if(DeltaZtoVtx>DeltaZ0Cut)continue;
```
Comment (rekovic) [inline L1Trigger/L1TTrackMatch/plugins/L1TrackerJetProducer.cc:170]: This if statement is now meaningful after DelatZ0Cut is assigned a value

**Comment by rekovic** [comment]
@emacdonald16 
Thanks for the fix in e1b9ec8.  
 * This fixes TrackerJets as now the DeltaZ0Cut is assigned 
a value and the if statement is meaningful (https://github.com/cms-l1t-offline/cmssw/pull/699/commits/e1b9ec886fb5930ee2ed1ca1ef380288c9847009#r201614928)
 * Cherry-picked it and tagged `1t-phase2-v2.16.11`

attn @mcepeda

**Comment by mcepeda** [comment]
@rekovic  - yes, @emacdonald16 mailed me yesterday with the information and after picking her latest commit I confirm now I have jets & mht in the menu trees 
(I'll move to 1t-phase2-v2.16.11 today)

**Pull request closed without merging**